The job **`aarch64-gnu-llvm-20-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/18241400327/job/51943709003) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/51943709003) [(plain)](https://github.com/rust-lang/rust/commit/958d9bb4fe980c550bcf259e3ffe811518660fa7/checks/51943709003/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
test [ui] tests/ui/zero-sized/zero-sized-btreemap-insert.rs ... ok

failures:

---- [ui] tests/ui/fn/invalid-sugg-for-unused-fn-arg-147303.rs stdout ----
Saved the actual stderr to `/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/fn/invalid-sugg-for-unused-fn-arg-147303/invalid-sugg-for-unused-fn-arg-147303.stderr`
diff of stderr:

1 error: unused variable: `rough`
-   --> $DIR/invalid-sugg-for-unused-fn-arg-issue-147303.rs:11:12
+   --> $DIR/invalid-sugg-for-unused-fn-arg-147303.rs:10:12
3    |
4 LL |     fn fun(rough: i32) {}
5    |            ^^^^^ help: if this is intentional, prefix it with an underscore: `_rough`

6    |
7 note: the lint level is defined here
-   --> $DIR/invalid-sugg-for-unused-fn-arg-issue-147303.rs:4:29
+   --> $DIR/invalid-sugg-for-unused-fn-arg-147303.rs:3:29
---
To only update this specific test, also pass `--test-args fn/invalid-sugg-for-unused-fn-arg-147303.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/fn/invalid-sugg-for-unused-fn-arg-147303.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/fn/invalid-sugg-for-unused-fn-arg-147303" "-A" "unused" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error: unused variable: `rough`
##[error]  --> /checkout/tests/ui/fn/invalid-sugg-for-unused-fn-arg-147303.rs:10:12
   |
LL |     fn fun(rough: i32) {} //~ERROR unused variable
   |            ^^^^^ help: if this is intentional, prefix it with an underscore: `_rough`
   |
note: the lint level is defined here
  --> /checkout/tests/ui/fn/invalid-sugg-for-unused-fn-arg-147303.rs:3:29
   |
LL | #![deny(unused_assignments, unused_variables)]
```

</details>

--- a/compiler/rustc_passes/src/liveness.rs
+++ b/compiler/rustc_passes/src/liveness.rs
@@ -1691,7 +1691,10 @@ impl<'tcx> Liveness<'_, 'tcx> {
                 if ln == self.exit_ln { false } else { self.assigned_on_exit(ln, var) };
 
             let mut typo = None;
-            for (hir_id, _, span) in &hir_ids_and_spans {
+            let filtered_hir_ids_and_spans = hir_ids_and_spans.iter().filter(|(hir_id, ..)| {
+                !matches!(self.ir.tcx.parent_hir_node(*hir_id), hir::Node::Param(_))
+            });
+            for (hir_id, _, span) in filtered_hir_ids_and_spans.clone() {
                 let ty = self.typeck_results.node_type(*hir_id);
                 if let ty::Adt(adt, _) = ty.peel_refs().kind() {
                     let name = Symbol::intern(&name);
@@ -1717,7 +1720,7 @@ impl<'tcx> Liveness<'_, 'tcx> {
                 }
             }
             if typo.is_none() {
-                for (hir_id, _, span) in &hir_ids_and_spans {
+                for (hir_id, _, span) in filtered_hir_ids_and_spans {
                     let ty = self.typeck_results.node_type(*hir_id);
                     // Look for consts of the same type with similar names as well, not just unit
                     // structs and variants.

--- a/tests/ui/fn/invalid-sugg-for-unused-fn-arg-147303.rs
+++ b/tests/ui/fn/invalid-sugg-for-unused-fn-arg-147303.rs
@@ -1,0 +1,13 @@
+// Regression test for <https://github.com/rust-lang/rust/issues/147303>.
+
+#![deny(unused_assignments, unused_variables)]
+
+mod m1 {
+    const _MAX_FMTVER_X1X_EVENTNUM: i32 = 0;
+}
+
+mod m2 {
+    fn fun(rough: i32) {} //~ERROR unused variable
+}
+
+fn main() {}

--- a/tests/ui/fn/invalid-sugg-for-unused-fn-arg-147303.stderr
+++ b/tests/ui/fn/invalid-sugg-for-unused-fn-arg-147303.stderr
@@ -1,0 +1,14 @@
+error: unused variable: `rough`
+  --> $DIR/invalid-sugg-for-unused-fn-arg-147303.rs:10:12
+   |
+LL |     fn fun(rough: i32) {}
+   |            ^^^^^ help: if this is intentional, prefix it with an underscore: `_rough`
+   |
+note: the lint level is defined here
+  --> $DIR/invalid-sugg-for-unused-fn-arg-147303.rs:3:29
+   |
+LL | #![deny(unused_assignments, unused_variables)]
+   |                             ^^^^^^^^^^^^^^^^
+
+error: aborting due to 1 previous error
+

--- a/tests/ui/lint/non-snake-case/lint-uppercase-variables.stderr
+++ b/tests/ui/lint/non-snake-case/lint-uppercase-variables.stderr
@@ -58,16 +58,7 @@ warning: unused variable: `Foo`
   --> $DIR/lint-uppercase-variables.rs:33:17
    |
 LL |     fn in_param(Foo: foo::Foo) {}
-   |                 ^^^
-   |
-help: if this is intentional, prefix it with an underscore
-   |
-LL |     fn in_param(_Foo: foo::Foo) {}
-   |                 +
-help: you might have meant to pattern match on the similarly named variant `Foo`
-   |
-LL |     fn in_param(foo::Foo::Foo: foo::Foo) {}
-   |                 ++++++++++
+   |                 ^^^ help: if this is intentional, prefix it with an underscore: `_Foo`
 
 error: structure field `X` should have a snake case name
   --> $DIR/lint-uppercase-variables.rs:10:5

